torch/cuda/__init__.py

```diff
@@ -15,6 +15,7 @@
 import traceback
 import warnings
 import threading
+from functools import lru_cache as _lru_cache
```

Comment:
Why rename?

Reply:
I want to be clear, that this is internal implementation detail and not expected to be used outside of this module, where `_` prefix is a pretty good indicator.

Reply:
But we don't do `import warnings as _warnings`, for example.

Reply:
Imo, this would be a good practice. Or use `__all__`